### DISCUSSION TITLE
Ghidra crash when analyze big mach-O binaries on debian (100 mb + files ?)

### DISCUSSION THREAD

**Discussion by ghost** [NationalSecurityAgency/ghidra#6550] [Q&A]
**Describe the bug**
when I run an analysis of a mach-O binary which is more than 100 mb on debian (same result on windows), ghidra crashes around the middle of the analysis (I tested with different mach-O binaries, I think it's because of their size because "small" binaries work well) You can see the log file : 
[logs.txt](https://github.com/NationalSecurityAgency/ghidra/files/15366178/logs.txt)

line 776 of log file : "./launch.sh: line 217: 15912 Killed"

**To Reproduce**
1. Add mach-O binary to project
2. Analyze it
3. Ghidra crash (only for me maybe idk)

**Expected behavior**
Complete analyze of the binary without crash

**Environment (please complete the following information):**
 - OS: [Debian & Windows]
 - Java Version: [17.0.10]
 - Ghidra Version: [11.0.4]
 - Ghidra Origin: [Official GitHub distro]

**Comment by seekbytes** [comment]
From logs it seems it’s being killed. Are you sure you haven’t hit the memory limits and then the OOM kills Ghidra?

**Comment by ghost** [comment]
Ok, how can I calculate the memory needed to analyze 100mb mach-O binary ?

**Comment by dev747368** [comment]
> Ok, how can I calculate the memory needed to analyze 100mb mach-O binary ?

It will depend on the specific analyzers that are involved in processing your binary.

You can open a memory usage window in Ghidra from the project window's Help menu.  For recent ghidra versions it will be Help|Runtime Info|Memory tab.  That will refresh every few seconds, so just leave it up in the background and if the used memory hits the maxmemory value and then you get a crash, its probably memory.

You can also disable individual analyzers before kicking off the analysis session.   I'm not familiar with all analyzers, but for example, the DWARF analyzer with extremely large binaries can need a bit of memory.

The name of the currently running analyzer will typically show in the bottom right corner of the tool window where there is a progress bar.  (sometimes... sometimes it might not be helpful)

**Comment by Karamann** [comment]
Ghidra's default memory settings might not be sufficient for analyzing large binaries. You can increase the heap size allocated to Ghidra by editing the launch.sh script (for Linux) or ghidra.ini file (for Windows). Look for the following lines and increase the values:

For launch.sh

`JAVA_OPTIONS="-Djava.net.useSystemProxies=true -Xmx4G -XX:MaxGCPauseMillis=200"
`

Change -Xmx4G to -Xmx8G or a higher value, depending on your system's available memory.

For ghidra.ini (Windows):

`-Xmx4G`
Change -Xmx4G to -Xmx8G or a higher value.
